internal/util/util.go

```diff
@@ -375,7 +375,7 @@ func GetVolumeSnapshotRestoreWithStatusData(restoreName string, PVCName string,
 		}
 
 		if len(vsrList.Items) > 0 {
```

Comment:
If user have made a restore before with the same name, it is a possibility that a vsr might exists from a different restore of the same name. Should we be using restore  UID instead/in addition to the restore name?

It is common in testing for cu to delete restore and try again and may got the wrong VSR for status.

Reply:
We will need a deleteItemAction plugin to handle deleting VSR when its restore is deleted. This is how we handle it for backup/VSB, which is outside of this plugin.

Reply:
Going to add logic to the VSB deleteItemAction to also delete corresponding VSR to handle such case, so will fix outside of this PR

Reply:
I still think adding UID would be a good idea. Similar to ownerReferences which has UID as well.